docs/source/serving/deploying_with_k8s.rst

```diff
@@ -0,0 +1,173 @@
+      name: hf-token-secret
+      namespace: llm
+    type: Opaque
+    data:
+      token: *******
+    ---
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: mistral-7b
+      namespace: llm
+      labels:
+        app: mistral-7b
+    spec:
+      replicas: 1
+      revisionHistoryLimit: 1
+      selector:
+        matchLabels:
+          app: mistral-7b
+      template:
+        metadata:
+          labels:
+            app: mistral-7b
+        spec:
+          volumes:
+          - name: cache-volume
+            persistentVolumeClaim:
+              claimName: mistral-7b
+          - name: shm
+            emptyDir:
+              medium: Memory
+              sizeLimit: "2Gi"
+          containers:
+          - name: mistral-7b
+            image: vllm/vllm-openai:latest
+            command: ["/bin/sh", "-c"]
+            args: [
+              "vllm serve mistralai/Mistral-7B-Instruct-v0.3 --trust-remote-code --enable-chunked-prefill --max_num_batched_tokens 1024"
+            ]
+            env:
+            - name: HUGGING_FACE_HUB_TOKEN
+              valueFrom:
+                secretKeyRef:
+                  name: hf-token-secret
+                  key: token
+            - name: VLLM_NO_USAGE_STATS
+              value: "1"
+            - name: DO_NOT_TRACK
+              value: "1"
+            ports:
+            - containerPort: 8000
+            resources:
+              limits:
+                cpu: "10"
+                memory: 20G
+                nvidia.com/gpu: "1"
+              requests:
+                cpu: "2"
+                memory: 6G
+                nvidia.com/gpu: "1"
+            volumeMounts:
+            - mountPath: /root/.cache/huggingface
+              name: cache-volume
+            - name: shm
+              mountPath: /dev/shm
+            livenessProbe:
+              httpGet:
+                path: /health
+                port: 8000
+              initialDelaySeconds: 60
+              periodSeconds: 10
+            readinessProbe:
+              httpGet:
+                path: /health
+                port: 8000
+              initialDelaySeconds: 60
+              periodSeconds: 5
+    ---
+
+2. **Create a Kubernetes Service for vLLM**
+
+Next, create a Kubernetes Service file to expose the `mistral-7b` deployment:
+
+.. code-block:: yaml
+
+    apiVersion: v1
+    kind: Service
+    metadata:
+      name: mistral-7b
+      namespace: llm
+    spec:
+      internalTrafficPolicy: Cluster
+      ipFamilies:
+      - IPv4
+      ipFamilyPolicy: SingleStack
+      ports:
+      - name: http-mistral-7b
+        port: 80
+        protocol: TCP
+        targetPort: 8000
+      selector:
+        app: mistral-7b
+      sessionAffinity: None
+      type: ClusterIP
+
+3. **Deploy and Test**
+
+Apply the deployment and service configurations using ``kubectl apply -f <filename>``:
+
+.. code-block:: console
+
+    kubectl apply -f deployment.yaml
+    kubectl apply -f service.yaml
+
```

Comment:
do we need port-forward here?

Reply:
There is no need to use port forwarding because I am accessing the LLM in the Kubernetes cluster via `mistral-7b.llm.svc.cluster.local`